scripts/main/photo.js

```diff
@@ -671,13 +671,12 @@ photo.setPublic = function (photoID, e) {
 				if (lychee.full_photo) {
 					$('.basicModal .choice input[name="full_photo"]').prop("checked", true);
 				}
-				// Photos shared individually are always hidden.
-				$('.basicModal .choice input[name="hidden"]').prop("checked", true);
+				if (lychee.public_photos_hidden) {
+					$('.basicModal .choice input[name="hidden"]').prop("checked", true);
+				}
 				if (lychee.downloadable) {
 					$('.basicModal .choice input[name="downloadable"]').prop("checked", true);
 				}
-				// Photos shared individually are always hidden.
-				$('.basicModal .choice input[name="hidden"]').prop("checked", true);
```

Comment:
Good optimisation! I'm unclear why that was there twice.

Reply:
I'm guessing it was a copy-paste error when share_button_visible was added.